### PR TITLE
feat: add StringUtil.java

### LINKED ISSUE #104: [P1] Add StringUtil.java

### PR DESCRIPTION
**Pull request by bezzang-dev**
**Added**
- `com/thesurvey/api/util/StringUtil.java` to reduce the repetitive code to trim String

**I changed PR and issue title `feat: add StringUtil.java and remove authentication parameter` to 
`feat: add StringUtil.java` because I reverted the incorrect refactoring.**

This PR closes #104 .

### CODE REVIEW

File: api/src/main/java/com/thesurvey/api/controller/AuthenticationController.java [hunk 1]
```diff
@@ -28,7 +28,7 @@ public AuthenticationController(AuthenticationService authenticationService) {
 
     @ApiOperation(value = "회원가입", notes = "회원가입을 요청합니다.")
     @ApiResponses(value = {
-        @ApiResponse(code = 200, message = "요청 성공"),
+        @ApiResponse(code = 201, message = "요청 성공"),
```
Comment (seiwonpark) [inline api/src/main/java/com/thesurvey/api/controller/AuthenticationController.java]: Please change this back to `200`.

File: api/src/main/java/com/thesurvey/api/controller/SurveyController.java [hunk 2]
```diff
@@ -71,17 +70,17 @@ public ResponseEntity<SurveyResponseDto> getSurvey(
 
     @ApiOperation(value = "설문조사 생성", notes = "새로운 설문조사를 생성합니다.")
     @ApiResponses(value = {
-        @ApiResponse(code = 200, message = "요청 성공"),
+        @ApiResponse(code = 201, message = "요청 성공"),
```
Comment (seiwonpark) [inline api/src/main/java/com/thesurvey/api/controller/SurveyController.java]: Please change this back to `200`.

File: api/src/main/java/com/thesurvey/api/service/AnsweredQuestionService.java [hunk 3]
```diff
@@ -50,8 +51,8 @@ public AnsweredQuestionService(SurveyRepository surveyRepository,
     }
 
     @Transactional
-    public void createAnswer(Authentication authentication,
-        AnsweredQuestionRequestDto answeredQuestionRequestDto) {
+    public void createAnswer(AnsweredQuestionRequestDto answeredQuestionRequestDto) {
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
```
Comment (seiwonpark) [inline api/src/main/java/com/thesurvey/api/service/AnsweredQuestionService.java]: What is the difference between sending Authentication parameter and getting authentication instance like on this code?

File: api/src/main/java/com/thesurvey/api/util/StringUtil.java [hunk 4]
```diff
@@ -0,0 +1,18 @@
+package com.thesurvey.api.util;
+
+import com.thesurvey.api.exception.BadRequestExceptionMapper;
+import com.thesurvey.api.exception.ErrorMessage;
+
+public class StringUtil {
+
+    public static String trim(String value) {
```
Comment (seiwonpark) [inline api/src/main/java/com/thesurvey/api/util/StringUtil.java:8]: Very nice!

File: api/src/main/java/com/thesurvey/api/util/UserUtil.java [hunk 5]
```diff
@@ -41,6 +41,9 @@ public static String getUserNameFromAuthentication(Authentication authentication
     }
 
     public static void validateUserAuthentication(Authentication authentication) {
+        if (authentication == null) {
```
Comment (seiwonpark) [inline api/src/main/java/com/thesurvey/api/util/UserUtil.java]: Could you merge this in one `if statement` with the below one by like using `&&` operator?

File: api/src/main/java/com/thesurvey/api/service/converter/CertificationTypeConverter.java [hunk 6]
```diff
@@ -1,39 +0,0 @@
-package com.thesurvey.api.service.converter;
-
-import com.thesurvey.api.domain.EnumTypeEntity.CertificationType;
-import java.util.List;
-import java.util.stream.Collectors;
-import org.springframework.context.annotation.Configuration;
-
-@Configuration
-public class CertificationTypeConverter {
```
Comment (seiwonpark) [inline api/src/main/java/com/thesurvey/api/service/converter/CertificationTypeConverter.java]: Why did you remove this file?

File: api/src/main/java/com/thesurvey/api/controller/AuthenticationController.java [hunk 7]
```diff
@@ -28,15 +29,15 @@ public AuthenticationController(AuthenticationService authenticationService) {
...
         @ApiResponse(code = 400, message = "잘못된 요청"),
         @ApiResponse(code = 404, message = "요청한 리소스 찾을 수 없음"),
         @ApiResponse(code = 500, message = "서버 내부 오류")
     })
     @PostMapping("/register")
     public ResponseEntity<UserResponseDto> register(
         @Valid @RequestBody UserRegisterRequestDto userRegisterRequestDto) {
-        return ResponseEntity.ok(authenticationService.register(userRegisterRequestDto));
+        return ResponseEntity.status(HttpStatus.CREATED).body(authenticationService.register(userRegisterRequestDto));
```
Comment (seiwonpark) [inline api/src/main/java/com/thesurvey/api/controller/AuthenticationController.java]: Please undo this job.

**Review by seiwonpark — CHANGES_REQUESTED** [review]
Please update your code and re-request.

File: api/src/main/java/com/thesurvey/api/controller/SurveyController.java [hunk 8]
```diff
@@ -71,17 +70,17 @@ public ResponseEntity<SurveyResponseDto> getSurvey(
...
         @ApiResponse(code = 403, message = "접근 권한 없음"),
         @ApiResponse(code = 404, message = "요청한 리소스 찾을 수 없음"),
         @ApiResponse(code = 500, message = "서버 내부 오류")
     })
     @PostMapping
-    public ResponseEntity<SurveyResponseDto> createSurvey(Authentication authentication,
+    public ResponseEntity<SurveyResponseDto> createSurvey(
         @Valid @RequestBody SurveyRequestDto surveyRequestDto) {
-        return ResponseEntity.ok(surveyService.createSurvey(authentication, surveyRequestDto));
+        return ResponseEntity.status(HttpStatus.CREATED).body(surveyService.createSurvey(surveyRequestDto));
```
Comment (seiwonpark) [inline api/src/main/java/com/thesurvey/api/controller/SurveyController.java]: Please change this back to `ok`.

File: api/src/main/java/com/thesurvey/api/service/AnsweredQuestionService.java (hunk 3 above)
Comment (seiwonpark) [reply api/src/main/java/com/thesurvey/api/service/AnsweredQuestionService.java]: I think these two are just the same. Maybe there might be some better way to resolve this.

File: api/src/main/java/com/thesurvey/api/service/converter/CertificationTypeConverter.java (hunk 6 above)
Comment (bezzang-dev) [reply api/src/main/java/com/thesurvey/api/service/converter/CertificationTypeConverter.java]: Since the converter only gets involved when retrieving data from the database and does not affect the storage of CertificationType in the database.. so I removed it

Comment (seiwonpark) [reply api/src/main/java/com/thesurvey/api/service/converter/CertificationTypeConverter.java]: Then, wouldn't this be much closer to `TypeUtil` class?

Comment (bezzang-dev) [reply api/src/main/java/com/thesurvey/api/service/converter/CertificationTypeConverter.java]: I will keep the file because it seems ambiguous whether this class is more suitable for `TypeUtil` or `Converter`..

**Review by seiwonpark — APPROVED** [review]
LGTM. Thanks!
